src/shared/udev-util.c

```diff
@@ -536,3 +538,20 @@ int udev_resolve_subsys_kernel(const char *string, char *result, size_t maxsize,
         }
         return 0;
 }
+
+int udev_queue_is_empty(void) {
+        return access("/run/udev/queue", F_OK) < 0;
```

Comment:
maybe take the opportunity here to fix error propagation on this?
i.e.

```c
int udev_queue_is_empty(void) {
        return access("/run/udev/queue") < 0 ? 
                (errno == ENOENT ? true : -errno) : false;
```

And then if the other errors really should be treated equal to "is empty" handle that in the caller?